core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsAddress.java

```diff
@@ -77,7 +77,7 @@ public void doWriteObject(ObjectOutput output, JGroupsAddress address) throws IO
       public JGroupsAddress doReadObject(ObjectInput unmarshaller) throws IOException, ClassNotFoundException {
          try {
             org.jgroups.Address address = org.jgroups.util.Util.readAddress(unmarshaller);
-            return (JGroupsAddress) JGroupsAddressCache.fromJGroupsAddress(address);
+            return (JGroupsAddress) JGroupsAddressCache.fromJGroupsAddressUnchecked(address);
```

Comment:
In theory, you're right. In practice, if you replace `fromJGroupsAddressUnchecked` with `fromJGroupsAddress`, you'll get an exception like this in the `LdapAuthenticationIT`:

```
org.infinispan.commons.CacheException: Unable to invoke method public void org.infinispan.marshall.core.GlobalMarshaller.start() on object of type GlobalMarshaller
Caused by: org.infinispan.commons.CacheException: Unable to invoke method public void org.infinispan.marshall.core.GlobalMarshaller.start() on object of type GlobalMarshaller
Caused by: java.lang.NoClassDefFoundError: org/jgroups/Address
	at org.infinispan.marshall.core.InternalExternalizers.load(InternalExternalizers.java:180)
	at org.infinispan.marshall.core.GlobalMarshaller.start(GlobalMarshaller.java:125)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.infinispan.commons.util.SecurityActions.lambda$invokeAccessibly$0(SecurityActions.java:91)
	... 89 more
Caused by: java.lang.ClassNotFoundException: org.jgroups.Address from [Module "deployment.a9b4f229-4004-47a6-9113-e651571940af.war:main" from Service Module Loader]
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:198)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:363)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:351)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:93)
	... 96 more
```

No idea why, but the method call does trigger the loading of `org.jgroups.Address`.